### ISSUE TITLE
LSI 9207-8i HBA Pci-e card is not detected in the pci_E3 slot of the MSI PRO Z690 Board with Dasharo 1.1.3

### ISSUE THREAD

**Issue by psmerica** [Dasharo/dasharo-issues#758]
### Device

MSI PRO Z690-A DDR4

### Dasharo version

V 1.1.3 and V 1.1.2

### Affected component(s) or functionality

The LSI 9207-8i HBA pci-e card is not detected on the pci_E3 slot of the board. After some troubleshooting it seems to not be detected on all of the pci-e slots that are provided from the z690 chipset. It does work on the pci-e1 slot that has its lanes provided by the cpu. This was tested both on Dasharo 1.1.3 and 1.1.2.

### Brief summary

LSI HBA card is no detected on pci_e lanes that are provided by the Z690 Chipset. The HBA card does work on the pci-e lanes provided from the CPU.

### How reproducible

The bug is reproducible every-time you try to use the HBA card in a slot that has its lanes provided by the Z690 chipset. It works every-time you plug it in a slot that has its lanes provided by the CPU.

### How to reproduce

1. Install LSI HBA card into pci-e slot pci_e3. 
2. Boot system.
3. LSI HBA not detected.

### Expected behavior

The LSI HBA is detected in the PCI_e3 slot.

### Actual behavior

The LSI HBA in not detected in the PCI_e3 slot.

**Comment by mkopec** [comment]
Could you run `sudo cbmem -c` on any Linux distro and attach the output here? That would help us a lot to find the cause for this issue.

Alternately you can also run the [HCL Report function in Dasharo Tools Suite](https://docs.dasharo.com/dasharo-tools-suite/documentation/#hcl-report), that might be easier for you

**Comment by miczyg1** [comment]
Also ` LSI 9207-8i ` is a PCIe x8 card, while all Z690A chipset provided PCIe slots are max x4... THe PCI_E1 is x16, that is why it probably work without problems...

**Comment by psmerica** [comment]
Yes its true the LSI 9207-8i is a x8 card and the pci-e3 slot is x4 lanes, but it still should work. I tested it with the stock MSI bios and it works great in that slot. At the very least it should be detected by the system, the card is not detected at all. This is not a problem of whether it works at full speed etc.. its a problem because it cannot be detected. I will provide logs later on after work.

**Comment by zirblazer** [comment]
I have a feeling this may be related to the IRQ problem that is known from issues trying to passthrough an USB Card, from same user:
https://github.com/Dasharo/dasharo-issues/issues/411  
There was also another user that in chat reported issues trying to passthrough a Video Card on Chipset slot and failed with QEMU throwing IRQ related error. So yeah, there is something definitely broken there.

**Comment by psmerica** [comment]
> Could you run `sudo cbmem -c` on any Linux distro (package coreboot-utils in most distros, or built from https://github.com/Dasharo/coreboot/tree/dasharo/util/cbmem) and attach the output here? That would help us a lot to find the cause for this issue.
> 
> Alternately you can also run the [HCL Report function in Dasharo Tools Suite](https://docs.dasharo.com/dasharo-tools-suite/documentation/#hcl-report), that might be easier for you

Here is the cbmem output and a HCL Report.

[cbmem.txt](https://github.com/Dasharo/dasharo-issues/files/14751314/cbmem.txt)
[Micro-Star_International_Co.,_Ltd._MS-7D25_Dasharo_(coreboot+UEFI)_v1.1.3_bcf92643-8efb-59b4-a1d5-18dc64fcbdf1_2024_03_26_01_30_51_630289367.tar.gz](https://github.com/Dasharo/dasharo-issues/files/14751315/Micro-Star_International_Co._Ltd._MS-7D25_Dasharo_.coreboot%2BUEFI._v1.1.3_bcf92643-8efb-59b4-a1d5-18dc64fcbdf1_2024_03_26_01_30_51_630289367.tar.gz)

**Comment by psmerica** [comment]
Here is the cbmem output and a HCL Report with the LSI card in the pci_e1 slot. The LSI card is now picked up by the pc.

[cbmeminpci1.txt](https://github.com/Dasharo/dasharo-issues/files/14751563/cbmeminpci1.txt)

[Micro-Star_International_Co.,_Ltd._MS-7D25_Dasharo_(coreboot+UEFI)_v1.1.3_1581e0d0-1b1c-5b9b-9e65-be0b36ec43f4_2024_03_26_02_12_33_969754216.tar.gz](https://github.com/Dasharo/dasharo-issues/files/14751565/Micro-Star_International_Co._Ltd._MS-7D25_Dasharo_.coreboot%2BUEFI._v1.1.3_1581e0d0-1b1c-5b9b-9e65-be0b36ec43f4_2024_03_26_02_12_33_969754216.tar.gz)

**Comment by miczyg1** [comment]
`[INFO ]  Found PCIe Root Port #5 at PCI: 00:1c.4.` this is the PCIE_3. SInce hotplug is not enabled, ti seems that the device was seen, because FSP left the PCIE Root Port enabled. But then it disappeared... :shrug:
